frontend/src/lib/components/details/person_media.svelte

```diff
@@ -1,40 +1,54 @@
 <script lang="ts">
-  import { fade } from "svelte/transition"
-  import InfiniteLoading from "svelte-infinite-loading"
+  import MediaCard from "$lib/components/media_card.svelte"
   import {
-    mediaIdToUrlConverter,
-    calculateAmountOfShownItems,
     removeDuplicates,
     sortListByPopularity,
     removeContentWithMissingImagePath,
-    getMediaTitle,
-  } from "../../utils"
-  import type { Media } from "../../types"
-  import { IMG_W342 } from "$lib/variables"
-  import { onMount } from "svelte"
+  } from "$lib/utils"
+  import type { Media } from "$lib/types"
+  import { PYTHON_API } from "$lib/variables"
+  import { currentCountry } from "$lib/stores/country"
+  import type { Meilisearch, Hit } from "$lib/generated"
+  import Spinner from "../loading/spinner.svelte"
 
   export let media: Media[]
 
-  let currentMediaAmount: number
-  let mediaStartAmount: number
+  let providerAmounts: number[]
 
-  // TODO: Kill with fire
-  onMount(() => {
-    currentMediaAmount = calculateAmountOfShownItems({
-      width: window.visualViewport.width,
-      xxl: 32,
-      xl: 28,
-      lg: 24,
-      md: 20,
-      sm: 16,
-      mobile: 10,
+  const hitProviderAmounts = (searchHits: Hit[]) => {
+    providerAmounts = []
+    searchHits.forEach(hit => {
+      let combinedAmount = 0
+      if (hit.providers) {
+        if ("flatrate" in hit.providers.results[$currentCountry]) {
+          combinedAmount += hit.providers.results[$currentCountry]["flatrate"].length
+        }
+        if ("free" in hit.providers.results[$currentCountry]) {
+          combinedAmount += hit.providers.results[$currentCountry]["free"].length
+        }
+      }
+      providerAmounts.push(combinedAmount)
     })
-    mediaStartAmount = currentMediaAmount
-  })
+  }
+
+  const lookupMedia = async (): Promise<Meilisearch> => {
+    const ids = media.map(v => v.id)
+    const res = await fetch(
+      `${PYTHON_API}/media?c=${$currentCountry}&limit=${ids.length}&ids=${ids.join(
+        "&ids="
+      )}`
+    )
+    const json: Meilisearch = await res.json()
+
+    hitProviderAmounts(json.hits)
+
+    return json
+  }
+
+  let refreshLookupMedia = lookupMedia
 
-  const loadMoreData = async ({ detail: { loaded } }) => {
-    currentMediaAmount += mediaStartAmount
-    loaded()
+  $: if ($currentCountry) {
+    refreshLookupMedia = lookupMedia
```

Comment:
Might be more descriptive of the "hack"

```suggestion
    refreshableLookupMedia = lookupMedia
```

Reply:
Deleted it after making it functional